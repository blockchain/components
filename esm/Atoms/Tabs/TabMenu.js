function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: flex-start;\n  align-items: center;\n  width: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  border-bottom: 2px solid ", ";\n  box-sizing: border-box;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
var Wrapper = styled.div(_templateObject(), function (props) {
  return props.theme['gray1'];
});
var Container = styled.div(_templateObject2(), function (props) {
  return props.width;
});

var TabMenu = function TabMenu(_ref) {
  var children = _ref.children,
      width = _ref.width;
  return React.createElement(Wrapper, null, React.createElement(Container, {
    width: width
  }, children));
};

TabMenu.propTypes = {
  children: PropTypes.node,
  width: PropTypes.string
};
TabMenu.defaultProps = {
  width: '100%'
};
export default TabMenu;