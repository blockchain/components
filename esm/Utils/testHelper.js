import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}