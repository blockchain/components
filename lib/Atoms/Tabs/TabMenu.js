"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: flex-start;\n  align-items: center;\n  width: ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: row;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  border-bottom: 2px solid ", ";\n  box-sizing: border-box;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents.default.div(_templateObject(), function (props) {
  return props.theme['gray1'];
});

var Container = _styledComponents.default.div(_templateObject2(), function (props) {
  return props.width;
});

var TabMenu = function TabMenu(_ref) {
  var children = _ref.children,
      width = _ref.width;
  return _react.default.createElement(Wrapper, null, _react.default.createElement(Container, {
    width: width
  }, children));
};

TabMenu.propTypes = {
  children: _propTypes.default.node,
  width: _propTypes.default.string
};
TabMenu.defaultProps = {
  width: '100%'
};
var _default = TabMenu;
exports.default = _default;