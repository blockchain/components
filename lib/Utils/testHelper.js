"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}