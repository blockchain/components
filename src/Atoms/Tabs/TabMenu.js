import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  border-bottom: 2px solid ${(props) => props.theme['tabMenuBorderColor']};
  box-sizing: border-box;
`
const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  width: ${(props) => props.width};
`

const TabMenu = ({ children, width }) => (
  <Wrapper>
    <Container width={width}>{children}</Container>
  </Wrapper>
)

TabMenu.propTypes = {
  children: PropTypes.node,
  width: PropTypes.string,
}

TabMenu.defaultProps = {
  width: '100%',
}

export default TabMenu
