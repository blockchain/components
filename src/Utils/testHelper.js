import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
