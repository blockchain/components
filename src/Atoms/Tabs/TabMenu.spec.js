import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import TabMenu from './TabMenu'

describe('TabMenu', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<TabMenu>MyTabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })

  it('should render correctly with height', () => {
    const tree = mountWithTheme(<TabMenu height="200px">MyTabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })

  it('should render correctly with toggled', () => {
    const tree = mountWithTheme(<TabMenu toggled>MyTabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })

  it('should render correctly with width', () => {
    const tree = mountWithTheme(<TabMenu width="50%">MyTabMenu</TabMenu>)

    expect(tree.find(TabMenu)).toMatchSnapshot()
  })
})
