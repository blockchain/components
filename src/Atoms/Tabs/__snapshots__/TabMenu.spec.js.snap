// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 3rem;
  border-bottom: 0.125rem solid #EEF2F5;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
  min-height: 100%;
}

@media (max-width:48rem) {
  .c1 > .disabled {
    display: none;
  }
}

@media (min-width:48rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<TabMenu
  height="3rem"
  toggled={false}
  width="100%"
>
  <styled.div
    height="3rem"
    width="100%"
  >
    <div
      className="c0"
      height="3rem"
      width="100%"
    >
      <styled.div
        toggled={false}
      >
        <div
          className="c1"
        >
          MyTabMenu
        </div>
      </styled.div>
    </div>
  </styled.div>
</TabMenu>
`;

exports[`TabMenu should render correctly with height 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 200px;
  border-bottom: 0.125rem solid #EEF2F5;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
  min-height: 100%;
}

@media (max-width:48rem) {
  .c1 > .disabled {
    display: none;
  }
}

@media (min-width:48rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<TabMenu
  height="200px"
  toggled={false}
  width="100%"
>
  <styled.div
    height="200px"
    width="100%"
  >
    <div
      className="c0"
      height="200px"
      width="100%"
    >
      <styled.div
        toggled={false}
      >
        <div
          className="c1"
        >
          MyTabMenu
        </div>
      </styled.div>
    </div>
  </styled.div>
</TabMenu>
`;

exports[`TabMenu should render correctly with toggled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 3rem;
  border-bottom: 0.125rem solid #EEF2F5;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
  min-height: 100%;
}

@media (max-width:48rem) {
  .c1 > .disabled {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:48rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<TabMenu
  height="3rem"
  toggled={true}
  width="100%"
>
  <styled.div
    height="3rem"
    width="100%"
  >
    <div
      className="c0"
      height="3rem"
      width="100%"
    >
      <styled.div
        toggled={true}
      >
        <div
          className="c1"
        >
          MyTabMenu
        </div>
      </styled.div>
    </div>
  </styled.div>
</TabMenu>
`;

exports[`TabMenu should render correctly with width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  min-height: 3rem;
  border-bottom: 0.125rem solid #EEF2F5;
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
  min-height: 100%;
}

@media (max-width:48rem) {
  .c1 > .disabled {
    display: none;
  }
}

@media (min-width:48rem) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<TabMenu
  height="3rem"
  toggled={false}
  width="50%"
>
  <styled.div
    height="3rem"
    width="50%"
  >
    <div
      className="c0"
      height="3rem"
      width="50%"
    >
      <styled.div
        toggled={false}
      >
        <div
          className="c1"
        >
          MyTabMenu
        </div>
      </styled.div>
    </div>
  </styled.div>
</TabMenu>
`;
